tests/rptest/tests/retention_policy_test.py

```diff
@@ -142,7 +142,8 @@ def __init__(self, test_context):
                              default_retention_segments * self.segment_size)
 
         si_settings = SISettings(test_context,
-                                 log_segment_size=self.segment_size)
+                                 log_segment_size=self.segment_size,
+                                 fast_uploads=True)
```

Comment:
Would we avoid this if we called `flush_manifest_clean_offset` in `garbage_collect()` after uploading the manifest? Otherwise this seems mildly concerning (though not necessarily blocking) since we're now delaying local GC.

Reply:
Hmm.  When I added this I was thinking of the test needing to see up to date manifests, but let me look into the impact on local GC...